pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java

```diff
@@ -411,6 +411,8 @@ public CompletableFuture<Void> closeAsync() {
             LOG.info("Closing PulsarService");
             state = State.Closing;
 
+            functionWorkerService.ifPresent(WorkerService::stop);
```

Comment:
It has been closed at https://github.com/apache/pulsar/blob/e1c0ad395c19c99369e733f44a2426de5c465787/pulsar-broker/src/main/java/org/apache/pulsar/PulsarBrokerStarter.java#L298

Reply:
That is cli code, not PulsarService, we also need to add close in PulsarService.

Reply:
What do you mean by cli code? I see that the functionWorkerService is passed to the PulsarService by here: https://github.com/apache/pulsar/blob/96930fda8a7776aaf6e61235a1d77ffe6e564052/pulsar-broker/src/main/java/org/apache/pulsar/PulsarBrokerStarter.java#L204

Reply:
@RobertIndie 

> What do you mean by cli code?

PulsarBrokerStarter belongs to cli code, it is called by using the `pulsar broker`. 

>  I see that the functionWorkerService is passed to the PulsarService by here:

We support passing the functionWorkerService to PulsarService, but the function worker should be closed by PulsarService, not PulsarBrokerStarter, which is a component of PulsarService.

If we don't close the functionWorkerService when running the close method of PulsarService, the functionWorkerService will break the PulsarService running, because the functionWorkerService cannot request the broker service, then call the PulsarService shutdown method. So when close and shutdown methods run together, the shutdown method will break close method.